pkg/test/assertions/assertions.go

```diff
@@ -20,10 +24,79 @@ import (
 //
 //	AssertThat(t, object, Is(Named("asdf")))
 //
-// Note that it intentionally doesn't accept a slice of predicates so that it is easy
-// to spot the failed predicate in the actual tests and the caller doesn't have to guess
-// which of the many supplied predicates might have failed.
-func AssertThat(t *testing.T, object client.Object, predicate Predicate[client.Object], msgAndArgs ...any) {
+// Note that this method accepts multiple predicates and reports any failures in them using
+// the Explain function.
+func AssertThat(t *testing.T, object client.Object, predicates ...Predicate[client.Object]) {
```

Comment:
Since you are changing a signature of an existing function let's be careful with breaking components which uses it. We need to get the other PRs ready or we can consider introducing a new function and keep the current one deprecated while updating all the dependencies.

Reply:
I have checked the code base in all repositories for this. Since this was introduced only very recently, it was used only on a couple of places, and none of them used the removed vararg parameter. So at this time,  this is a "source compatible" change.

Reply:
Great. Thanks for checking.